starter-code/scripts/controllers/articleController.js

```diff
@@ -45,6 +49,7 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //upon page load .loadAll sets the ctx.articles to be the same as the already populated arry of all object instances. It passes the context onto the next function where it calls the .index function which contains the articleView.index function; rendering all the articles to the view.
```

Comment:
This is not quite what is happening! Look more carefully at the beginning of the articleController.loadByCategory function definition -- it is not setting ctx.articles to articlesinCategory. It is simply declaring a function and assigning it to the variable articleData. Nothing is being called at this point, and the ctx object is not being altered until articleData is invoked. 

There is a crucial conditional statement present - if we have an already populated articles array, then we don't have to worry about doing anything more and we **can** simply add an articles property to the ctx object with the value of the Articles.allArticle array. Then, the next function in the middleware chain is called. If we don't have any articles yet (on initial pageload), we will call Article.fetchAll and pass in the function that we defined, articleData, so that we can retrieve our articles from our JSON file via AJAX or from local storage, save a reference to them as Article.allArticles, and THEN call our callback (which attach them to the context object and calls the next function in the middleware chain).